platform-hub-web/src/app/projects/services/project-service-selector-popup.html

```diff
@@ -0,0 +1,79 @@
+<md-dialog aria-label="Choose a project service">
+  <md-toolbar class="md-toolbar-tools md-accent">
+    <h2 ng-if="$ctrl.serviceIsOptional">
+      Choose a project or a service from a project
+    </h2>
+    <h2 ng-if="!$ctrl.serviceIsOptional">
+      Choose a service from a project
+    </h2>
+    <span flex></span>
+    <md-button
+      class="close-button md-icon-button"
+      ng-click="$ctrl.cancel()"
+      aria-label="Close dialog">
+      <md-icon>clear</md-icon>
+    </md-button>
+  </md-toolbar>
+
+  <loading-indicator loading="$ctrl.loading"></loading-indicator>
+
+  <md-dialog-content class="md-dialog-content" layout="column" layout-padding>
+
+    <form name="$ctrl.form">
+      <md-input-container class="md-block">
+        <label for="project">Project:</label>
+        <md-select
+          name="project"
+          ng-model="$ctrl.selectedProject"
+          ng-disabled="!$ctrl.Projects.all.length"
+          ng-change="$ctrl.handleProjectChanged()">
+          <md-option
+            ng-repeat="p in $ctrl.Projects.all"
+            ng-value="p">
+            {{p.shortname}}
+            ({{p.name}})
+          </md-option>
+        </md-select>
+      </md-input-container>
+
+      <md-input-container class="md-block">
+        <label for="service">Service:</label>
+        <md-select
+          name="service"
+          ng-model="$ctrl.selectedService"
```

Comment:
I think this should also be `required` depending on `serviceIsOptional`.

Reply:
I could add that in, but the form level validation is not used in this component – instead we explicitly check the `selectedService` field to enable/disable button.

Mind you, it would provide a nicer UI (highlighted in red if not filled in), so I'll add that.